src/d3a/models/strategy/home_meter.py

```diff
@@ -532,7 +538,7 @@ def _one_sided_market_event_tick(self, market, offer=None):
         """
         if not self.state.can_buy_more_energy(market.time_slot):
             return
-        if not offer and not market.offers:
+        if not offer or not market.offers:
```

Comment:
@fievelk IMO this was a bug. It revealed itself in the `Strategies follow yearly user profile saved in the DB` integration test. Do you agree on this change or was the `and` added on purpose?

Reply:
I'm not actually confident about this, since these `if`s are quite convoluted. This line was implemented after this comment: https://github.com/gridsingularity/d3a/pull/1159#discussion_r655984021
The original version was taken from the load implementation: https://github.com/gridsingularity/d3a/blob/153648206fcca5ccc86b3aa47e265ccaee01b1db/src/d3a/models/strategy/load_hours.py#L266-L274

We don't want to exit here if `offer` is `None`, because in that case we would never execute this line:
```python
acceptable_offer = offer if offer else self._find_acceptable_offer(market)
```
So it looks like we only want to exit if we have neither `offer` nor `market.offers`, because in that case we couldn't find an acceptable offer anywhere.
If we have at least one of them (either `offer` or `market.offers`) we need to go on and check if there's an `acceptable_offer` in the markets.

Reply:
I think i understand. Let me check what went wrong there.

Reply:
Actually, I found that the if statements are not translated correctly following the comment from Spyros.
This should be correct now:
https://github.com/gridsingularity/d3a/pull/1152/commits/316229f891477d722acf2512785273c5c547bd4c#diff-1187ba841fc3bdd8c4b0200dc3033dfc48bd59ba5c22485e605fd0d1293e1d78R541

In order to also catch the case when the offer is indeed None.